Source/JavaScriptCore/dfg/DFGStoreBarrierInsertionPhase.cpp

```diff
@@ -431,17 +435,50 @@ class StoreBarrierInsertionPhase : public Phase {
                 // So this analysis must mark an allocation that escapes to the heap as being part of the primordial
                 // epoch.
 
-                ClobberSet writes = writeSet(m_graph, m_node);
-                if (writes.overlaps(Heap) || writes.overlaps(Stack)) {
-                    auto escape = [] (Node* node) {
+                auto readFunc = [&] (const AbstractHeap& heap) {
+                    if (heap.kind() == Stack) {
+                        Vector<AbstractHeap, 4> toRemove;
+                        for (auto [trackedHeap, node] : potentialStackEscapes) {
+                            if (trackedHeap.overlaps(heap)) {
+                                node->setEpoch(Epoch());
+                                toRemove.append(trackedHeap);
+                            }
+                        }
+
+                        for (const auto& heap : toRemove)
+                            potentialStackEscapes.remove(heap);
```

Comment:
Let's use `potentialStackEscapes.removeIf` since it is more efficient.

Reply:
Sounds good!